docs/adr/adr-004-state-interaction.md

```diff
@@ -0,0 +1,173 @@
+# ADR #004: Celestia-Node State Interaction for March 2022 Testnet
+
+<hr style="border:3px solid gray"> </hr>
+
+## Authors
+
+@renaynay
+
+## Changelog
+
+* 2022-01-14: initial draft
+
+<hr style="border:2px solid gray"> </hr>
+
+## Context
+
+Currently, celestia-node lacks a way for users to submit a transaction to celestia-core.
+
+## Decision
+
+Both celestia **light** and **full** nodes will run `StateService`. 
+`StateService` will be responsible for lifecycle of the given implementation of `StateAccessor`.
+
+`StateAccessor` will be an interface for interacting with celestia-core in order to retrieve account-related information
+as well as submit transactions.
+
+### `StateService`
+
+`StateService` will manage the lifecycle of the `StateAccessor` and also expose several higher-level RPC endpoints for 
+users to access the methods provided by the `StateAccessor`.
+
+```go
+type StateService struct {
+    accessor StateAccessor
+}
+``` 
+
+### `StateAccessor`
+
+`StateAccessor` is defined by the two methods listed below and may be expanded in the future to include more methods 
+related to accounts and transactions in later iterations.
+
+```go
+type StateAccessor interface {
+    func AccountBalance(context.Context, Account) (Balance, error)
+    func SubmitTx(context.Context, Tx) error
+}
+```
+
+`StateAccessor` will have 3 separate implementations under the hood: 
+1. **DIRECT**: direct RPC connection with a celestia-core endpoint handed to the node upon initialisation
+(**required for this iteration**) 
+2. **P2P**: discovery of a **bridge** node, and then establish direct RPC connection (*nice-to-have for this iteration*)
```

Comment:
No RPC in p2p impl:
* Unsecure,
	* With RPC you connect without a secured tunnel and proper pubkey handshake, just with an address
* We will need to rewrite it once 'local' is implemented.
	* Because 'local' won't be sharing state over Tendermint's RPC
* Ugly, the code will be super ugly to make this work. Especially addresses convertions. 
* We will waste more time making this work and fixing all the caveats, while simple req/resp with libp2p is straightforward. IDK, why this frightens everybody like something hard when it's not. It won't be any different from HeaderExchange besides discovery which we need anyway

Reply:
resolved?

Reply:
Yes